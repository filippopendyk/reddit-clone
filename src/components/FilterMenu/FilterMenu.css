.activeFilter{
    background-color: #4864F5;
    color: white;
}

.filter-menu{
    display: flex;
    align-items: center;
    overflow:hidden;
}

.filter-menu button{
    padding: 0.75rem 4rem;
    font-size: 1.15rem;
    font-weight: bold;
    border: 0;
    flex-grow: 1;
}

