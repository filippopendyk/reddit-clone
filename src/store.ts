import { configureStore } from "@reduxjs/toolkit";
import filterSlice from "./features/filter/filterSlice";
import { postsSliceReducer } from "./features/posts/postsSlice";
import { fetchMostPopularCategories, topCategoriesReducer } from "./features/topCategories/topCategoriesSlice";

const store = configureStore({
    reducer: {
        topCategories: topCategoriesReducer,
        posts: postsSliceReducer,
        filter: filterSlice,
    },
});

store.dispatch(fetchMostPopularCategories());

export default store;

// Define types for usedispatch and useselector

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch;