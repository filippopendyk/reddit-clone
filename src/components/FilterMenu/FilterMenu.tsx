import { setFilterAsBest, setFilterAsDefault, setFilterAsNew, setFilterAsTop } from "../../features/filter/filterSlice";
import { useAppDispatch, useAppSelector } from "../../hooks";
import './FilterMenu.css';

const FilterMenu: React.FC = () => {
    const filter = useAppSelector((state) => state.filter.filter);
    const dispatch = useAppDispatch();
    console.log(filter);

    const handleBest = (event: React.MouseEvent<HTMLButtonElement>) :void => {
        event.preventDefault();
        dispatch(setFilterAsBest());
    }

    const handleTop = (event: React.MouseEvent<HTMLButtonElement>) :void => {
        event.preventDefault();
        dispatch(setFilterAsTop());
    } 

    const handleNew = (event: React.MouseEvent<HTMLButtonElement>) :void => {
        event.preventDefault();
        dispatch(setFilterAsNew());
    }

    return (
        <div className="filter-menu">
            <button className={filter === 'best' ? 'activeFilter' : undefined} onClick={handleBest}>Best</button>
            <button className={filter === 'top' ? 'activeFilter' : undefined} onClick={handleTop}>Top</button>
            <button className={filter === 'new' ? 'activeFilter' : undefined} onClick={handleNew}>New</button>
        </div>
    )
};

export default FilterMenu;