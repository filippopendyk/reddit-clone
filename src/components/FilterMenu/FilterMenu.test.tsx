import FilterMenu from "./FilterMenu";
import renderer from 'react-test-renderer';
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import store from "../../store";

describe('FilterMenu', () => {
    it('Renders without crashing', () => {
        const tree = renderer
        .create(
            <Provider store={store}>
                <FilterMenu/>
            </Provider>
        )
        .toJSON();
        expect(tree).toMatchSnapshot();
    })
})