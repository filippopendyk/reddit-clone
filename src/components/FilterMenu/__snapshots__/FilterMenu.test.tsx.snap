// Vitest Snapshot v1

exports[`FilterMenu > Renders without crashing 1`] = `
<div
  className="filter-menu"
>
  <button
    className="activeFilter"
    onClick={[Function]}
  >
    Best
  </button>
  <button
    onClick={[Function]}
  >
    Top
  </button>
  <button
    onClick={[Function]}
  >
    New
  </button>
</div>
`;
